crates/bevy_math/src/curve/easing.rs

```diff
@@ -605,6 +631,382 @@ pub enum EaseFunction {
+pub struct CircularInOut;
+
+/// `f(t) ≈ 2.0^(10.0 * (t - 1.0))`
+///
+/// The precise definition adjusts it slightly so it hits both `(0, 0)` and `(1, 1)`:
+/// `f(t) = 2.0^(10.0 * t - A) - B`, where A = log₂(2¹⁰-1) and B = 1/(2¹⁰-1).
+///
+#[doc = include_str!("../../images/easefunction/ExponentialIn.svg")]
+#[derive(Copy, Clone)]
+pub struct ExponentialIn;
+
+/// `f(t) ≈ 1.0 - 2.0^(-10.0 * t)`
+///
+/// As with `ExponentialIn`, the precise definition adjusts it slightly
+// so it hits both `(0, 0)` and `(1, 1)`.
+///
+#[doc = include_str!("../../images/easefunction/ExponentialOut.svg")]
+#[derive(Copy, Clone)]
+pub struct ExponentialOut;
+
+/// Behaves as `ExponentialIn` for t < 0.5 and as `ExponentialOut` for t >= 0.5
+///
+#[doc = include_str!("../../images/easefunction/ExponentialInOut.svg")]
+#[derive(Copy, Clone)]
+pub struct ExponentialInOut;
+
+/// `f(t) = -2.0^(10.0 * t - 10.0) * sin((t * 10.0 - 10.75) * 2.0 * π / 3.0)`
+///
+#[doc = include_str!("../../images/easefunction/ElasticIn.svg")]
+#[derive(Copy, Clone)]
+pub struct ElasticIn;
+
+/// `f(t) = 2.0^(-10.0 * t) * sin((t * 10.0 - 0.75) * 2.0 * π / 3.0) + 1.0`
+///
+#[doc = include_str!("../../images/easefunction/ElasticOut.svg")]
+#[derive(Copy, Clone)]
+pub struct ElasticOut;
+
+/// Behaves as `ElasticIn` for t < 0.5 and as `ElasticOut` for t >= 0.5
+///
+#[doc = include_str!("../../images/easefunction/ElasticInOut.svg")]
+#[derive(Copy, Clone)]
+pub struct ElasticInOut;
+
+/// `f(t) = 2.70158 * t³ - 1.70158 * t²`
+///
+#[doc = include_str!("../../images/easefunction/BackIn.svg")]
+#[derive(Copy, Clone)]
+pub struct BackIn;
+
+/// `f(t) = 1.0 +  2.70158 * (t - 1.0)³ - 1.70158 * (t - 1.0)²`
+///
+#[doc = include_str!("../../images/easefunction/BackOut.svg")]
+#[derive(Copy, Clone)]
+pub struct BackOut;
+
+/// Behaves as `BackIn` for t < 0.5 and as `BackOut` for t >= 0.5
+///
+#[doc = include_str!("../../images/easefunction/BackInOut.svg")]
+#[derive(Copy, Clone)]
+pub struct BackInOut;
+
+/// bouncy at the start!
+///
+#[doc = include_str!("../../images/easefunction/BounceIn.svg")]
+#[derive(Copy, Clone)]
+pub struct BounceIn;
+
+/// bouncy at the end!
+///
+#[doc = include_str!("../../images/easefunction/BounceOut.svg")]
+#[derive(Copy, Clone)]
+pub struct BounceOut;
+
+/// Behaves as `BounceIn` for t < 0.5 and as `BounceOut` for t >= 0.5
+///
+#[doc = include_str!("../../images/easefunction/BounceInOut.svg")]
+#[derive(Copy, Clone)]
+pub struct BounceInOut;
+
+/// `n` steps connecting the start and the end. Jumping behavior is customizable via
+/// [`JumpAt`]. See [`JumpAt`] for all the options and visual examples.
+#[derive(Copy, Clone)]
+pub struct Steps(usize, JumpAt);
+
+/// `f(omega,t) = 1 - (1 - t)²(2sin(omega * t) / omega + cos(omega * t))`, parametrized by `omega`
+///
+#[doc = include_str!("../../images/easefunction/Elastic.svg")]
+#[derive(Copy, Clone)]
+pub struct Elastic(f32);
+
+// Implements `Curve<f32>` by calling a function in `easing_functions`.
```

Comment:
Is this supposed to be a doc comment?

```suggestion
/// Implements `Curve<f32>` by calling a function in `easing_functions`.
```